dev-support/release-vm/README.md

```diff
@@ -0,0 +1,63 @@
+# HBase Release Env
+
+This is a vagrant project that provides a virtual machine environment suitable
+for running an Apache HBase release.
+
+Requires:
+* VirtualBox
+* Vagrant
+* The private portion of your signing key avilable in the local GPG agent
+* The private portion of your Github authentication key available in the local GPG agent
+
+## Usage
+
+Unlock the local keyring before proceeding (this should prompt you for your GPG passphrase). For
+example, assuming you have an authentication key configured in your keyring, this will do the
+trick.
+
+```sh
+$ ssh -T git@github.com
+Hi <you>! You've successfully authenticated, but GitHub does not provide shell access.
+```
+
+Export the public portion of your signing credentials where the guest VM can access it,
+
+```sh
+$ gpg --export <you>@apache.org > gpg.<you>.apache.pub
+```
+
+Launch the VM
+
+```sh
+$ vagrant up
+```
+
+Connect to the VM. This should forward your ssh- and gpg-agent session.
+
+```sh
+$ vagrant ssh
+```
+
+Now from the VM, import your identity and verify the gpg-agent passthrough is working correctly.
+
+```sh
+$ gpg --no-autostart --import /vagrant/gpg.<you>.apache.pub
+...
+gpg: Total number processed: 1
+gpg:               imported: 1
+$ gpg --no-autostart --detach --armor --sign repos/hbase/pom.xml
```

Comment:
I had to resupply my password for my gpg key here. I didn't expect that. I even logged out, reset my gpg agent with 

gpgconf --kill all && gpg-connect-agent /bye

... signed something off the vm (which required me to fill in my password)..... I then did vagrant ssh and had to provide password again.

Reply:
I didn't expect to have to reprovide my gpg password if forwarding going on.

Reply:
Signing a new file, I did not have to provide my password... which is good. I have a vm up that does not have my private key which is good.....

Reply:
If agent forwarding is working, and your password expire time is set reasonably long, you shouldn't need to enter the password on the guest side. Something is not working properly.

Did you `vagrant destroy -f` your VM before applying my latest changes? Specifically, you'd need to re-run the puppet provisioner in order to get the new ssh configuration.

It's also worth confirming that there is no gpg-agent process on the guest OS, ever. As I understand it, the gpg process run in the foreground of the guest terminal should always be communicating back to the gpg-agent process running on the host via the socket. I don't know how to ensure the process is not launched on the guest, other than via the `--no-autostart` argument added to the commands.

Reply:
I destroyed the old vm before running this test but did it in virutalbox, not with vagrant destroy. I just tried it and this time I did not have to supply my key password testing signing up in the vm.